oozie/src/main/java/org/apache/falcon/workflow/engine/OozieWorkflowEngine.java

```diff
@@ -974,30 +965,61 @@ private void performAction(String cluster, JobAction action, CoordinatorAction c
         }
     }
 
-    private void reRunCoordAction(String cluster, CoordinatorAction coordinatorAction) throws FalconException {
+    public CoordinatorAction.Status reRunCoordAction(String cluster, CoordinatorAction coordinatorAction,
+                                                      Properties props, boolean isForced) throws FalconException {
         try {
             OozieClient client = OozieClientFactory.get(cluster);
+            if (props == null) {
+                props = new Properties();
+            }
+            // In case if both props exists one should be removed otherwise it will fail.
+            // This case will occur when user runs workflow with skip-nodes property and
+            // try to do force rerun or rerun with fail-nodes property.
+            if (props.containsKey(OozieClient.RERUN_FAIL_NODES)
+                    && props.containsKey(OozieClient.RERUN_SKIP_NODES)) {
+                LOG.warn("Both " + OozieClient.RERUN_SKIP_NODES + " and " + OozieClient.RERUN_FAIL_NODES
+                        + " are present in workflow params removing" + OozieClient.RERUN_SKIP_NODES);
+                props.remove(OozieClient.RERUN_SKIP_NODES);
```

Comment:
Suppose a user do force re-run once then RERUN_FAIL_NODES will be set  and he again doing rerun by setting other property RERUN_SKIP_NODES, it leads to rerun in failure, that's why one property was removed to make rerun successful. Or vice versa will happen first user rerun by passing some props and he did force rerun it might not work. Thats why we are removing one